java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -757,6 +760,94 @@ public static Table concatenate(Table... tables) {
+   * Example:
+   *
+   *  Single column:
+   *      idx            0   1   2   3   4
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 1 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  3 }
+   * NaNs in column values do not work especially
+   * if the table has multiple columns.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be of mixed data types and
+   * the result gives the index of the value pertaining to the matching type column
+   * in the input table.
+   * @param areNullsSmallest true if nulls are assumed smallest
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @return ColumnVector with lower bound indices for all rows in valueTable
+   */
+  public ColumnVector lowerBound(boolean areNullsSmallest,
+      Table valueTable, boolean[] descFlags) {
+    assertTableChecks(valueTable);
+    return new ColumnVector(gdfBound(this.nativeHandle, valueTable.nativeHandle,
+      descFlags, areNullsSmallest, false));
+  }
+
+  /**
+   * Given a sorted table return the upper bound.
+   * Example:
+   *
+   *  Single column:
+   *      idx            0   1   2   3   4
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 3 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  5 }
+   * NaNs in column values do not work especially
+   * if the table has multiple columns.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be mixed and the result gives the index
+   * of the value pertaining to the matching type column in the input table.
+   * @param areNullsSmallest true if nulls are assumed smallest
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @return ColumnVector with upper bound indices for all rows in valueTable
+   */
+  public ColumnVector upperBound(boolean areNullsSmallest,
+      Table valueTable, boolean[] descFlags) {
+    assertTableChecks(valueTable);
```

Comment:
There are a lot of checks from this call that are not described in the javadocs.  The fact that an empty input table is not allowed scares me a bit because it feels like a valid use case that might hurt end users. 

Also it would be good to have a follow on issue filed to fix the bounds checks so an empty table behaves like we would expect it to.

Reply:
https://github.com/rapidsai/cudf/pull/3229/files#diff-98c23212fa6fcce605601c29795bb281R80
helps the concern we have. I will open an issue as follow on once this goes in.